src/persistence/SubjectExecutor.ts

```diff
@@ -240,7 +240,8 @@ export class SubjectExecutor {
                 this.queryRunner.broadcaster.broadcastBeforeInsertEvent(
                     result,
                     subject.metadata,
-                    subject.entity!,
+                    subject.entity ??
+                        subject.createValueSetAndChangeMapsWithoutValues()[0],
```

Comment:
It is hard to understand what is createValueSetAndChangeMapsWithoutValues() index zero. I mean, that's open source. How would you feel, if you find stuff like that?